hack/release-stable-minor.sh

```diff
@@ -70,7 +71,7 @@ touch "${STABILIZATION_LOG}"
 # Execute stabilization-changes.py until it stops making changes
 while true; do
 	echo "Running stabilization-changes.py: iteration ${I}" | tee -a "${STABILIZATION_LOG}"
-	if ! "${DIR}"/stabilization-changes.py &>> "${STABILIZATION_LOG}"; then
+	if ! "${DIR}"/stabilization-changes.py >> "${STABILIZATION_LOG}" 2>&1 ; then
```

Comment:
Bash 3.2 is almost 20 years old now. TBH I am not sure whether we really need to keep our script able to run on _default_ MacOS? It is a while but IIRC it was possible to install more recent bash on Mac with `brew`?

Reply:
Our script runnable on MacOS will make my work easy.
I thought a small change like this to keep the compatible would not be a big deal. 
It can still run with Bash in Linux box.

But yes, if I can fix the thing by brew, I do not need the change.
It seems that brew can: https://formulae.brew.sh/formula/bash
Let me try it out.

Reply:
Seems working, together with the help of changing the Shebang.

```console
$ bash --version
GNU bash, version 5.2.37(1)-release (aarch64-apple-darwin24.2.0)
```

I hope changing Shebang is fine to you.
From reading https://www.cyberciti.biz/tips/finding-bash-perl-python-portably-using-env.html, and
https://stackoverflow.com/questions/10376206/what-is-the-preferred-bash-shebang
i feel `#!/usr/bin/env bash` (over `#!/bin/sh`) should be even our default choice for portability of the shell script.
However, it seems it is the other way around in our repo (maybe that is not intensional?):
```console
$ rg 'bin/sh'
hack/exposure-length.sh
1:#!/bin/sh

hack/release-stable-minor.sh
1:#!/bin/sh

hack/release-ga.sh
1:#!/bin/sh

hack/release-open.sh
1:#!/bin/sh
```

Reply:
I could run with `bash hack/release-stable-minor.sh 4.18` without changing the shebang as a workaround in MacOS.
But for my learning, is it a good practice using `#!/usr/bin/env bash` instead in your opinion?

I could also use `alias sed=gsed` for my shell so the change on env. var. could go away too.
Let me know your preference.

Reply:
Yeah I'm definitely not against making scripts convenient to run for you and others with Macs, just feel like "our scripts should only use Bash 3 features" is a tradeoff that would go too far. IIRC installing modern bash with `brew` is just a very mild one-time inconvenience which to me seems acceptable for the ability to use modern features in our scripts.

Shebang change is absolutely fine (I also find it preferable to plain `#!/bin/sh`), and the swappable `sed` too.